toolz/functoolz.py

```diff
@@ -200,6 +200,7 @@ def __init__(self, *args, **kwargs):
 
         self.__doc__ = getattr(func, '__doc__', None)
         self.__name__ = getattr(func, '__name__', '<curry>')
+        self.__module__ = getattr(func, '__module__', None)
```

Comment:
We probably want #335.  This was also explored and discussed in #230.  My bad for letting things linger.

Things can get convoluted, though, so I prefer straightforward and explicit even if it's a little more verbose.  For example, `__annotations__` will need to be custom-handled for `curry` much like `curry.__signature__`.

To your point, though, I'll go ahead and add `__qualname__` to `curry` in Python 3.